Created workflow config

BX326 opened the pull request:
I've set up a config for GitHub Actions. Currently, the CI workflow is triggered upon pull request creation and update. If we want, we can utilize ``matrix`` (as shown [here](https://github.com/marketplace/actions/set-up-a-rust-toolchain)) to test additional Rust toolchains on other platforms (e.g. Windows and macOS).

BX326 committed:
Created workflow config

BX326 committed:
Specified Rust toolchain

philipwastakenwastaken commented:
Great PR. I think it makes a lot of sense to utilize  `matrix` to test against different platforms. None of the work we are doing is inherently platform specific, so testing against all 3 major OS (Linux, macOS, WIndows) should not be an issue. 

I am open to merging after `matrix`is introduced.

philipwastakenwastaken commented on .github/workflows/url2ref.yml:
No issues as far as I can see.

BX326 committed:
Added matrix strategy

BX326 committed:
Bash as default in npm step

Some commands in the npm build scripts are incompatible with Powershell.

BX326 committed:
Added build script for npm

The npm build script fails on a Windows platform due to the use of forward slashes in the paths -- even when bash is specified as the default shell in the action config. Therefore, the npm build script has been extracted to its own bash script.


Co-authored-by: philipwastakenwastaken <37301670+philipwastakenwastaken@users.noreply.github.com>
